cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1591,7 +1618,62 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
         }
 
         Future<ReconciliationState> kafkaServiceEndpointReady() {
-            return withVoid(serviceOperations.endpointReadiness(namespace, kafkaService, 1_000, operationTimeoutMs));
+            Future<ReconciliationState> fut = withVoid(serviceOperations.endpointReadiness(namespace, kafkaService, 1_000, operationTimeoutMs));
+            Future<ReconciliationState> blockingFuture = Future.future();
+
+            fut.compose(v -> {
+                vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
```

Comment:
Whether it's needed or not, FYI there's `serviceOperations.getAsync()`, which would mean you didn't need to do the work within `createSharedWorkerExecutor` here if `CrdOperations.updateStatus()` was async (i.e. returned `Future`). If `CrdOperations` only has a sync version they you end up forcing any async callers to use `createSharedWorkerExecutor`. It's better to have a single async method in `CrdOperations`. You might not need a sync version there at all.